velox/common/memory/tests/MemoryPoolTest.cpp

```diff
@@ -2167,6 +2167,120 @@ TEST(MemoryPoolTest, debugMode) {
   EXPECT_EQ(allocRecords.size(), 0);
 }
 
+TEST(MemoryPoolTest, debugModeWithFilter) {
+  constexpr int64_t kMaxMemory = 10 * GB;
+  constexpr int64_t kNumIterations = 100;
+  const std::vector<int64_t> kAllocSizes = {128, 8 * KB, 2 * MB};
+  const std::vector<bool> debugEnabledSet{true, false};
+  for (const auto& debugEnabled : debugEnabledSet) {
+    MemoryManager manager{
+        {.capacity = kMaxMemory, .debugEnabled = debugEnabled}};
+
+    // leaf child created from MemoryPool, not match filter
+    MemoryPoolImpl::setDebugPoolNameRegex("NO-MATCH");
+    auto root0 = manager.addRootPool("root0");
+    auto pool0 = root0->addLeafChild("PartialAggregation.0.0");
+    auto* buffer0 = pool0->allocate(1 * KB);
+    EXPECT_TRUE(std::dynamic_pointer_cast<MemoryPoolImpl>(pool0)
+                    ->testingDebugAllocRecords()
+                    .empty());
+    pool0->free(buffer0, 1 * KB);
+
+    // leaf child created from MemoryPool, match filter
+    MemoryPoolImpl::setDebugPoolNameRegex(".*PartialAggregation.*");
+    auto root1 = manager.addRootPool("root1");
+    auto pool1 = root1->addLeafChild("PartialAggregation.0.1");
+    auto* buffer1 = pool1->allocate(1 * KB);
+    if (!debugEnabled) {
+      EXPECT_EQ(
+          std::dynamic_pointer_cast<MemoryPoolImpl>(pool1)
+              ->testingDebugAllocRecords()
+              .size(),
+          0);
+    } else {
+      EXPECT_EQ(
+          std::dynamic_pointer_cast<MemoryPoolImpl>(pool1)
+              ->testingDebugAllocRecords()
+              .size(),
+          1);
+    }
+    pool1->free(buffer1, 1 * KB);
+
+    // old pool should not be affected by updated filter
+    buffer0 = pool0->allocate(1 * KB);
+    EXPECT_TRUE(std::dynamic_pointer_cast<MemoryPoolImpl>(pool0)
+                    ->testingDebugAllocRecords()
+                    .empty());
+    pool0->free(buffer0, 1 * KB);
+
+    // leaf child created from MemoryPool, match filter
+    MemoryPoolImpl::setDebugPoolNameRegex(".*OrderBy.*");
+    auto root2 = manager.addRootPool("root2");
+    auto pool2 = root2->addLeafChild("OrderBy.0.0");
+    auto* buffer2 = pool2->allocate(1 * KB);
+    if (!debugEnabled) {
+      EXPECT_EQ(
+          std::dynamic_pointer_cast<MemoryPoolImpl>(pool2)
+              ->testingDebugAllocRecords()
+              .size(),
+          0);
+    } else {
+      EXPECT_EQ(
+          std::dynamic_pointer_cast<MemoryPoolImpl>(pool2)
+              ->testingDebugAllocRecords()
+              .size(),
+          1);
+    }
+    pool2->free(buffer2, 1 * KB);
+
+    // leaf child created from aggr MemoryPool, match filter
+    auto intPool = root2->addAggregateChild("AGG-Pool");
+    auto pool9 = intPool->addLeafChild("OrderBy.0.1");
```

Comment:
nit: why this is pool9?

Reply:
It was later added. Let me refactor the naming.